src/StructuredDiscussions/TopicRepository.php

```diff
@@ -49,4 +49,28 @@ public function getByTitle( Title $title ): ?SDTopic {
 
 		return new SDTopic( $viewTopic['topic'] );
 	}
+
+	public function getByOwner( Title $title ): array {
+		$parameters = [ 'page' => $title->getTalkPage()->getFullText() ];
+		$topics = $this->callSubmodule( 'view-topiclist', $parameters );
+
+		$rtr = [];
+
+		if ( $topics === null ) {
+			return $rtr;
+		}
+
+		if ( !isset( $topics['topiclist'] ) ) {
+			return $rtr;
+		}
+
+		foreach ( $topics[ 'topiclist' ][ 'roots' ] as $root ) {
+			$copy = $topics[ 'topiclist' ];
+			$copy[ 'roots' ] = [ $root ];
+
+			$rtr []= new SDTopic( $copy );
+		}
+
+		return $rtr;
```

Comment:
```suggestion
		return array_map(function ( array $root ) use ( $topics ) {
			$copy = $topics['topiclist'];
			$copy['roots'] = [ $root ];
			
			return new SDTopic( $copy );
		}, $topics['topiclist']['roots'] );
```

Reply:
What is the added benefit of using the array_map?
All I can see it does is increase memory usage.

Reply:
It is functional :)

But there is no real added benefit, so feel free to ignore.